modules/odata/odata-core/src/main/java/org/eclipse/dirigible/engine/odata2/sql/processor/ResultSetReader.java

```diff
@@ -42,12 +42,28 @@ protected Map<String, Object> getEntityDataFromResultSet(SQLSelectBuilder select
     }
 
 
+    protected ResultSetEntity getResultSetEntityForView(SQLSelectBuilder selectEntityQuery, final EdmEntityType entityType,
```

Comment:
ForView is misleading, because yo can have generated key also for table. May be it will be better to remove getResultSetEntityForView and getResultSetEntity and rename getResultSetEntityForObject to get ResultSetEntity with last input parameter renamed from isView to hasGeneratedId. And just remove the if statement from AbstractSQLProcessor to use this new method.

Reply:
Fixed in [b6ad0d1](https://github.com/eclipse/dirigible/pull/1709/commits/b6ad0d13aca48d01ba9430b4a4b0ebab3778d71c)